### ISSUE TITLE
monarch-data-load non-responsive when building

### ISSUE THREAD

**Issue by jnguyenx** [monarch-initiative/monarch-devops#32]
Already discussed on Skype, but filing a ticket to keep a reference.

**Comment by jnguyenx** [comment]
Interesting message when I logged in the machine:

jnguyenxuan@monarch-access:~$ ssh monarch-data-load
Welcome to Ubuntu 14.04.4 LTS (GNU/Linux 3.19.0-49-generic x86_64)

 * Documentation:  https://help.ubuntu.com/

** System information disabled due to load higher than 12.0**

Last login: Wed Mar  2 15:31:06 2016 from monarch4-2.cslu.ohsu.edu

**Comment by TomConlin** [comment]
possibly related or split off to new ticket.
suggest non-attended jobs (jenkins, cron...) 
be niced to reduce impact on interactive use

**Comment by jnguyenx** [comment]
There's currently a heavy job running on Jenkins. There are multiple issues, in addition to the one mentioned in https://github.com/monarch-initiative/monarch-devops/issues/32#issuecomment-191495515.

1. ssh-ing to monarch-access takes a while (~2 minutes).
2. scigraph-data-prod and scigraph-data-dev becomes also non-responsive.
3. Cannot open http://vm-monitor.monarchinitiative.org/ganglia to see the state of the VMs.

My wild guess is that since all these VMs sit on the same baremetal machine, there is too much disk I/O going on for the baremetal machine to handle.

@vanmatre can you verify the disk usage (and other relevant metrics) of monarch4 for today between 3pm and 5pm please?

**Comment by jnguyenx** [comment]
I managed to access ganglia for a minute, monarch4 is very red:

![screenshot from 2016-05-13 16-47-33](https://cloud.githubusercontent.com/assets/4312088/15264672/c25d7d6e-192a-11e6-989b-de2fd95e5da4.png)

@vanmatre what would be a good fix? Redistribute the VMs to make sure that jenkins never sits with a production VM? Or pump up the jenkins VM?

**Comment by vanmatre** [comment]
I'm on the bus now so I cannot do much....

 The load average is inthe 70's I'll take a look to see which vm is bring it down. It appears to be swapping so we likely have an over commitment of memory.

Which VM do you think is doing it?



Ethan VanMatre
Informatics Research Analyst
Institute on Development and Disability
Oregon Health & Science University
CSLU - GH40
3181 SW Sam Jackson Park Rd
Portland, OR 97239
(503) 346-3764
vanmatre@ohsu.edu
________________________________

**Comment by jnguyenx** [comment]
It's definitely the one with jenkins (monarch-data-load).

**Comment by vanmatre** [comment]
You want me to kill the java job?



Ethan VanMatre
Informatics Research Analyst
Institute on Development and Disability
Oregon Health & Science University
CSLU - GH40
3181 SW Sam Jackson Park Rd
Portland, OR 97239
(503) 346-3764
vanmatre@ohsu.edu
________________________________

**Comment by jnguyenx** [comment]
No please don't, it's close to finishing. We just need to think about a
long term solution.

**Comment by vanmatre** [comment]
OK sounds good to me. Think about it over the weekend on how to throttle it to keep the load more reasonable. I'll think of creative solutions.

--Ethan




Ethan VanMatre
Informatics Research Analyst
Institute on Development and Disability
Oregon Health & Science University
CSLU - GH40
3181 SW Sam Jackson Park Rd
Portland, OR 97239
(503) 346-3764
vanmatre@ohsu.edu
________________________________

**Comment by jnguyenx** [comment]
@vanmatre did you have the time to think of a solution?

I was thinking about two options, we can get faster disk (SSDs?) or we can install jenkins on a baremetal machine instead of a VM to have a better I/O.

I'll be available for the rest for the week to solve this critical issue, just ping me on Skype or by email.

**Comment by vanmatre** [comment]
Working with Jeremy I was able to find a solution to this problem. Under heavy disk load the VM would become non-responsive. The VM's load average was low and the hardware load average would go up around 100.

By changing the disk cache mode and passing IO directly to the base hardware we were able to see the VM and hardware run as we would expect. The VM load average topped out around 20 with 15 being more likely and the hardware load average running 3 or so.

The VM responded as one would expect a system doing heavy lifting and continued to be responsive.

Looks like we have a solution.
